producer.go

```diff
@@ -71,6 +86,34 @@ func (p *segmentProducer) newWriter(cfg Config) error {
 	return nil
 }
 
+// onMessageDelivery is a callback function for kafka-go's writer
+// which is calling the ack functions for the messages which were
+// (successfully or unsuccessfully) delivered to Kafka.
+func (p *segmentProducer) onMessageDelivery(messages []kafka.Message, err error) {
+	if len(messages) == 0 {
+		return
+	}
+	for _, m := range messages {
+		// accessed also when in Send(), when messages need to be sent
+		p.m.Lock()
+		ackFunc, ok := p.ackFuncs[p.getID(m)]
+		delete(p.ackFuncs, p.getID(m))
+		p.m.Unlock()
+
+		if !ok {
+			// todo we probably need something better
+			// this will the kafka writer panic too, which will terminate the whole program
+			panic(fmt.Errorf("ack func for %v not registered", p.getID(m)))
+		}
```

Comment:
I'm not happy with this part. I'm looking for a better way for the plugin to let Conduit know, that it encountered an error, and not simply kill itself. This is happening in a different goroutine, so I'm looking for a nice way to "teleport" the error.